src/engine/filters/enginefilteriir.h

```diff
@@ -25,13 +25,11 @@ enum IIRPass {
     IIR_HP2,
 };
 
-
-class EngineFilterIIRBase : public EngineObjectConstIn {
+class EngineFilterIIRBase : public EngineObjectConstIn { // clazy:exclude=missing-qobject-macro
```

Comment:
I'm questioning the purpose of this class. We could just merge it into `EngineFilterIIR`:

```
$ git grep EngineFilterIIRBase
src/analyzer/analyzerwaveform.h:class EngineFilterIIRBase;
src/analyzer/analyzerwaveform.h:    EngineFilterIIRBase* m_filter[FilterCount];
src/engine/filters/enginefilteriir.h:class EngineFilterIIRBase : public EngineObjectConstIn {
src/engine/filters/enginefilteriir.h:class EngineFilterIIR : public EngineFilterIIRBase {
```

Reply:
Ah, it's because EngineFilterIIR is a template class.

Reply:
Anyway, we can't use it here because there is no CPP file for it and leads to linker errors:

```
mold: error: undefined symbol: EngineFilterIIRBase::qt_metacast(char const*)
>>> referenced by enginefilterlinkwitzriley4.cpp
>>>               libmixxx-lib.a(enginefilterlinkwitzriley4.cpp.o):(EngineFilterLinkwitzRiley4Low::qt_metacast(char const*))>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o):(EngineFilterBessel4Low::qt_metacast(char const*))>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o):(EngineFilterBessel4Band::qt_metacast(char const*))>>> referenced 13 more times

mold: error: undefined symbol: typeinfo for EngineFilterIIRBase
>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o)>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o)>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o)>>> referenced 7 more times

mold: error: undefined symbol: EngineFilterIIRBase::staticMetaObject
>>> referenced by enginefilterlinkwitzriley4.cpp
>>>               libmixxx-lib.a(enginefilterlinkwitzriley4.cpp.o)>>> referenced by enginefilterlinkwitzriley4.cpp
>>>               libmixxx-lib.a(enginefilterlinkwitzriley4.cpp.o)>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o)>>> referenced 13 more times

mold: error: undefined symbol: EngineFilterIIRBase::qt_metacall(QMetaObject::Call, int, void**)
>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o):(EngineFilterBessel4Low::qt_metacall(QMetaObject::Call, int, void**))>>> referenced by enginefilterlinkwitzriley4.cpp
>>>               libmixxx-lib.a(enginefilterlinkwitzriley4.cpp.o):(EngineFilterLinkwitzRiley4Low::qt_metacall(QMetaObject::Call, int, void**))>>> referenced by enginefilterbessel4.cpp
>>>               libmixxx-lib.a(enginefilterbessel4.cpp.o):(EngineFilterBessel4Band::qt_metacall(QMetaObject::Call, int, void**))>>> referenced 13 more times

clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
good point, QObjects can't be templated...